setup.sh

```
@@ -77,10 +77,28 @@ else
     success=false
 fi
 
-# Activate pnpm (version set in package.json)
-corepack enable pnpm || success=false
+# Activate pnpm via corepack, falling back to npm install if corepack is missing
+echo -e "${YELLOW}Setting up pnpm...${NC}"
+if command -v corepack &>/dev/null; then
+    corepack enable pnpm || success=false
+elif command -v npm &>/dev/null; then
+    echo -e "${YELLOW}corepack not found — installing pnpm via npm...${NC}"
+    npm install -g corepack && corepack enable pnpm || {
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)